gluoncv/data/transforms/presets/center_net.py

```diff
@@ -0,0 +1,265 @@
+        img = mx.nd.image.to_tensor(img)
+        img = mx.nd.image.normalize(img, mean=mean, std=std)
+        tensors.append(img.expand_dims(0))
+        origs.append(orig_img)
+    if len(tensors) == 1:
+        return tensors[0], origs[0]
+    return tensors, origs
+
+def load_test(filenames, short, max_size=1024, mean=(0.485, 0.456, 0.406),
+              std=(0.229, 0.224, 0.225)):
+    """A util function to load all images, transform them to tensor by applying
+    normalizations. This function support 1 filename or iterable of filenames.
+
+    Parameters
+    ----------
+    filenames : str or list of str
+        Image filename(s) to be loaded.
+    short : int
+        Resize image short side to this `short` and keep aspect ratio.
+    max_size : int, optional
+        Maximum longer side length to fit image.
+        This is to limit the input image shape. Aspect ratio is intact because we
+        support arbitrary input size in our SSD implementation.
+    mean : iterable of float
+        Mean pixel values.
+    std : iterable of float
+        Standard deviations of pixel values.
+
+    Returns
+    -------
+    (mxnet.NDArray, numpy.ndarray) or list of such tuple
+        A (1, 3, H, W) mxnet NDArray as input to network, and a numpy ndarray as
+        original un-normalized color image for display.
+        If multiple image names are supplied, return two lists. You can use
+        `zip()`` to collapse it.
+
+    """
+    if isinstance(filenames, str):
+        filenames = [filenames]
+    imgs = [mx.image.imread(f) for f in filenames]
+    return transform_test(imgs, short, max_size, mean, std)
+
+
+class CenterNetDefaultTrainTransform(object):
+    """Default SSD training transform which includes tons of image augmentations.
+
+    Parameters
+    ----------
+    width : int
+        Image width.
+    height : int
+        Image height.
+    num_class : int
+        Number of categories
+    scale_factor : int, default is 4
+        The downsampling scale factor between input image and output heatmap
+    mean : array-like of size 3
+        Mean pixel values to be subtracted from image tensor. Default is [0.485, 0.456, 0.406].
+    std : array-like of size 3
+        Standard deviation to be divided from image. Default is [0.229, 0.224, 0.225].
+    """
+    def __init__(self, width, height, num_class, scale_factor=4, mean=(0.485, 0.456, 0.406),
+                 std=(0.229, 0.224, 0.225), **kwargs):
+        self._kwargs = kwargs
+        self._width = width
+        self._height = height
+        self._num_class = num_class
+        self._scale_factor = scale_factor
+        self._mean = np.array(mean, dtype=np.float32).reshape(1, 1, 3)
+        self._std = np.array(std, dtype=np.float32).reshape(1, 1, 3)
+        self._data_rng = np.random.RandomState(123)
+        self._eig_val = np.array([0.2141788, 0.01817699, 0.00341571],
+                                 dtype=np.float32)
+        self._eig_vec = np.array([
```

Comment:
where can I find reference of this matrix? As it is similar but not identical with https://github.com/apache/incubator-mxnet/blob/master/src/io/image_aug_default.cc#L545

Reply:
It's origin is hard to track now, maybe from Corner implementation. The values are calculated using COCO dataset statistics but I guess it's rather subtle for training on difference dataset. I can take a look and see if the eigenvalues/eigenvector are important in the future